tests/conftest.py

```diff
@@ -0,0 +1,14 @@
+import pytest
+
+
+@pytest.fixture
+def isolated_test_env(tmp_path):
+    project_dir = tmp_path / "project"
+    global_dir = tmp_path / "global"
+    project_dir.mkdir()
+    global_dir.mkdir()
+
+    return {
+        "project_dir": project_dir,
+        "global_dir": global_dir
+    }
```

Comment:
Thanks for keeping the globa vs. project configs here...
I believe that for the purpose of integration testing we could simplify this to just use the 
`tmp_path / zowe.config.json`
```suggestion
    config_path = Path(__file__).resolve().parent.parent.parent / "zowe.config.json"
    config_file = tmp_path / "zowe.config.json"
    config_file.write_text(config_path.read_text())
    return tmp_path
```

Reply:
I think instead of `tmp_path` we need to return the `config_file` and `config_path` variables from the function. It would look something like this:
```py
from pathlib import Path

import pytest


@pytest.fixture
def isolated_test_env(tmp_path):
    config_path = Path(__file__).resolve().parent.parent.parent / "zowe.config.json"

    config_file = tmp_path / "zowe.config.json"
    config_file.write_text(config_path.read_text())

    return {
        "config_path": config_path, 
        "config_file": config_file
    }
```

And for the other file (`tests/integration/test_integration_env.py`):
```py
def test_isolated_env_does_not_touch_real_home(isolated_test_env):
    config_path = isolated_test_env["config_path"]
    config_file = isolated_test_env["config_file"]

    assert config_path.exists(), f"Missing config at {config_path}"
    assert config_file.exists()
    assert "profiles" in config_file.read_text()
```

And yes this simplifies the overall structure by isolating all the logic and assertions in different functions🫡. Please let me know if I am mistaken anywhere...

Reply:
If this is alright then we can possibly move towards making the structure more modular by using it like this: `class TestIsolatedEnv(unittest.TestCase):` as you suggested. A step by step approach would be better so I would like to first make a commit for modifications as mentioned above then move towards the modular integration...

Reply:
@zFernand0 is it ok to require `zowe.config.json` to be at the root level of the project, or should we allow it to reside at any folder level?